packages/frontend-app-api/src/wiring/createPluginInfoAttacher.ts

```diff
@@ -0,0 +1,249 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ConfigApi } from '@backstage/core-plugin-api';
+import {
+  FrontendFeature,
+  FrontendPluginInfo,
+} from '@backstage/frontend-plugin-api';
+import { OpaqueFrontendPlugin } from '@internal/frontend';
+import { JsonObject, JsonValue } from '@backstage/types';
+import once from 'lodash/once';
+// Avoid full dependency on catalog-model
+// eslint-disable-next-line @backstage/no-relative-monorepo-imports
+import {
+  parseEntityRef,
+  stringifyEntityRef,
+} from '../../../catalog-model/src/entity/ref';
+
+/**
+ * A function that resolves plugin info from a plugin manifest and package.json.
+ *
+ * @public
+ */
+export type FrontendPluginInfoResolver = (ctx: {
+  packageJson?(): Promise<JsonObject>;
+  manifest?(): Promise<JsonObject>;
+  defaultResolver(overrides?: {
+    packageJson?: JsonObject;
+    manifest?: JsonObject;
+  }): Promise<{ info: FrontendPluginInfo }>;
+}) => Promise<{ info: FrontendPluginInfo }>;
+
+export function createPluginInfoAttacher(
+  config: ConfigApi,
+  infoResolver: FrontendPluginInfoResolver = ctx => ctx.defaultResolver(),
+): (feature: FrontendFeature) => FrontendFeature {
+  const applyInfoOverrides = createPluginInfoOverrider(config);
+
+  return (feature: FrontendFeature) => {
+    if (!OpaqueFrontendPlugin.isType(feature)) {
+      return feature;
+    }
+
+    const plugin = OpaqueFrontendPlugin.toInternal(feature);
+
+    return {
+      ...plugin,
+      info: once(async () => {
+        const manifestLoader = plugin.infoOptions?.manifest;
+        const packageJsonLoader = plugin.infoOptions?.packageJson;
+
+        const { info: resolvedInfo } = await infoResolver({
+          manifest: manifestLoader,
+          packageJson: packageJsonLoader,
+          async defaultResolver(overrides) {
+            const manifest = overrides?.manifest ?? (await manifestLoader?.());
+            const packageJson =
+              overrides?.packageJson ?? (await packageJsonLoader?.());
+
+            return {
+              info: {
+                ...resolvePackageInfo(packageJson),
+                ...resolveManifestInfo(manifest),
+              },
+            };
+          },
+        });
+
+        const infoWithOverrides = applyInfoOverrides(plugin.id, resolvedInfo);
+        return normalizePluginInfo(infoWithOverrides);
+      }),
+    };
+  };
+}
+
+function normalizePluginInfo(info: FrontendPluginInfo) {
+  return {
+    ...info,
+    ownerEntityRefs: info.ownerEntityRefs?.map(ref =>
+      stringifyEntityRef(
+        parseEntityRef(ref, {
+          defaultKind: 'group',
```

Comment:
```suggestion
          defaultKind: 'Group',
```

doesn't really matter, but felt good having set right :)